### ISSUE TITLE
Application to Neural Network Policy

### ISSUE THREAD

**Issue by AlbertoSinigaglia** [ssokota/mmd#2]
Hi there, great paper, just finished reading it... one question about this codebase, where is (if exists) the implementation with a DeepNN for the policy? 

I happened to see the video of your presentation on YouTube where you say that this can be applied to large scale problems, and also already red the #1 issue, yet I'm not really sure about how to apply this to a large scale setting.

Is there ay blogpost about this with more implementation detail? 

I can only find this part of the code
https://github.com/ssokota/mmd/blob/62bc8a22aeb887de65ed8205307314f8303480c6/mmd/efg/learners.py#L125-L126

that seems relevant, but it's using a tabular setting, and a "list of infostates", which is not as scalable as I expected, and neither I see any reference about this on your colleague repo, so I'm probably missing something (sorry for that...)

**Comment by ssokota** [comment]
Hi, this repository doesn't include any code for training with neural network policies. The issue you referenced contains the modifications to RLlib that were used in the paper to do training with neural network policies.

**Comment by AlbertoSinigaglia** [comment]
@ssokota Hi, thank you so much for the reply.

As I'm currently cleating a Self-play RL agent for my thesis on a large scale imperfect information game, I wanted to implement this to try it out, as the final PPO agent that I have trained still have some exploitability, testing it with DQN as best response.
I've also tried adversarial training, and even though it helped, the best response still have a positive reward in expectation. The only piece that I'm missing that I don't see in the paper, is the definition of the magnet policy.

I see in the paper you refer multiple times a uniform probability, however, I see here for example that it's not the case

https://github.com/ssokota/mmd/blob/62bc8a22aeb887de65ed8205307314f8303480c6/mmd/nfg/learners.py#L75-L79

Is the magnet some sort of EMA in the case of DeepNN, uniform or what?

Thanks again
